### PR TITLE
Basic testing infrastructure for starlight

### PR DESCRIPTION
**Pull request by dwebchapey**
Closes issue #75 .

This is a basic testing Infrastructure using mocha testing framework for starlight to make sure AST and code generation is exercised correctly. A random contract from `./test/contracts` is picked and tested. Further tests can be added later to the framework as and when needed. 

Currently the following checks are done .

-  AST check for circuits 
- AST check for contracts
- AST check for orchestration
- Check whether circuits are created for all functions in AST
- Check whether orchestration files are created for all functions in AST.

The AST check is performed by comparing with a tested zapp structure in `./truezapps` folder.

To test , run `npm test` from root 

An option is added to commander

```
 .option(
    '-m, --modify <modify>',
    'Ovewrite the file in truezapps folder',
  );

```

which is created with an intention to modify the AST strcuture `./truezapps` in case of AST structure breaking changes. Logic can be added to trigger this during original zappification of zol.
Additionally ,  the AST JSON file in orchestration  is renamed to `${options.inputFileName}_ast.json` to have consistency across languages.

### CODE REVIEW

File: src/tests.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      if(options.modifyAST == 'y')
+      {
+      changesets.applyChangeset(PreGeneratedASTJsonObject, VariyingData);
+      fs.writeFileSync('./truezapps/'+options.inputFileName+'/'+ASTType+'/'+options.inputFileName+'_ast.json', JSON.stringify(PreGeneratedASTJsonObject, null, 4), 'utf8');
+      logger.info('AST file mddified');
+      return 'false';    
+      }
+    }
+    else
+      return 'false';
```
Comment (MirandaWood) [inline src/tests.ts:34]: Should this be `true`? As I think we reach here if the hashes match after doing a `JSON.stringify`

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import { checkASThashes, checkCodeGen, checktestASTExists } from '../built/tests.js'
+import mkdirs from '../bin/mkdirs.mjs';
+import zappify from '../built/index.js'; 
+var files = fs.readdirSync('./test/contracts/')
+let chosenFile = files[Math.floor(Math.random() * files.length)]
+const inputFilePath = './test/contracts/'+chosenFile;
+const modifyAST = 'n';
+const inputFileName = path.parse(inputFilePath).name;
+// commander converts 'zapp-name' to 'zappName'
+const zappName = inputFileName;
```
Comment (MirandaWood) [inline test/test.js:14]: I like the random zapp! Could we also add a line that logs which zapp is being tested? Something like `logger.info('Testing ${zappName}')`

File: src/tests.ts (hunk 1 above)
Comment (dwebchapey) [reply src/tests.ts:34]: Right . We reach here when the ASTs match after trimming the extra whitespaces. Should we accept that as true?

File: test/test.js (hunk 2 above)
Comment (dwebchapey) [reply test/test.js:14]: Sure . I will log few more bits of information.

File: src/tests.ts (hunk 1 above)
Comment (MirandaWood) [reply src/tests.ts:34]: Yes I think so - the main information in the AST would remain the same so that counts as a pass imo

**Review by MirandaWood — APPROVED** [review]

**Review by SwatiEY — APPROVED** [review]

**Pull request merged**